selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/core/UrlUtil.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.core;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/** @author Alexander Garagatyi */
+public class UrlUtil {
+
+  /**
+   * Creates {@link URL} using provided parameters.
+   *
+   * @throws IllegalArgumentException in case URL creation fails because of provided parameters
+   */
+  public static URL url(String protocol, String host, int port, String path)
+      throws IllegalArgumentException {
+    if ((80 == port && "http".equals(protocol)) || (443 == port && "https".equals(protocol))) {
```

Comment:
What are the benefits?

Reply:
Of what?

Reply:
Of having url `http://localhost/api` in comparison with `http://localhost:80/api`?

Anyway, my concern is about using magic constants :)
If would refine `if` clause:

```
int httpsDefaultPort = new URL("https://"+host).getDefaultPort();
int httpDefaultPort = new URL("http://"+host").getDefaultPort();

if ((httpDefaultPort == port && "http".equals(protocol)) || (httpsDefaultPort == port && "https".equals(protocol))) {
...
}
```

Reply:
I'm afraid it may lead to some weird errors when URL after passing from one component to another may lose default port and some error may happen in tests. It is also weird to see a URL with default port in it. By the way, Che itself would not use the port in URL in this case so it is more consistent.

I don't like magic constants too, but this tool can be used extensively and building URL is quite an expensive operation that involves parsing and validation. I do not think it is a good thing to do on each call.
But I can create constant fields such as:
```
private static final String DEFAULT_HTTP_PORT = 80;
private static final String DEFAULT_HTTPS_PORT = 443;
```
And use it in code. Would it be OK for you?

Reply:
Ok